matrix-room.c

```diff
@@ -865,6 +865,35 @@ void matrix_room_handle_timeline_event(PurpleConversation *conv,
     g_free(escaped_body);
 }
 
+PurpleConversation *matrix_room_join_conversation(
```

Comment:
As per my comments above, I think this us unnecessary. But even if it's not, most if this is C&Ped from matrix_room_create_conversation. Please try to avoid c&ping code.